controllers/common_test.go

```diff
@@ -148,4 +158,42 @@ var _ = Describe("GetGeneratedJobName", func() {
 			Expect(uidWithoutHyphens).To(ContainSubstring(generatedJobName))
 		})
 	})
+
+	When("An additional postfix is required and maxNameLen is sufficiently large", func() {
+		BeforeEach(func() {
+			maxNameLen = 253
+			objectMetaName = "object.meta.name"
+			generatedPostfix = "generated.postfix"
+			shouldJobNameHavePostfix = true
+
+		})
+
+		It("Concatenates the name, shortened uid, generatedPostfix", func() {
+			Expect(generatedJobName).To(ContainSubstring(objectMetaName))
+			Expect(generatedJobName).To(ContainSubstring(uidWithoutHyphens))
+			Expect(generatedJobName).To(ContainSubstring(generatedPostfix))
```

Comment:
Why are we differentiating between different substrings of the `required` parameter? Shouldn't the function's behavior be the same for the entire `required` parameter?

Reply:
Could we also verify the correct ordering for all the values and that they are delimited in the correct places?

Reply:
I think its still a good idea to have required = postfix+uid in order to check that if maxlen<required, its the uid part that is returned.

Reply:
We can do the additional tests Nick suggests, but given that we don't have any requirement on the ordering I think we are good as is.